src/utils/blockedMonitor.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { AzureFunctionsRpcMessages as rpc } from './../../azure-functions-language-worker-protobuf/src/rpc';
+import LogCategory = rpc.RpcLog.RpcLogCategory;
+import LogLevel = rpc.RpcLog.Level;
+import blockedAt = require('blocked-at');
+
+export function startBlockedMonitor(channel: { log: (log: rpc.IRpcLog) => void }, threshold = 100): NodeJS.Timer {
+    let blockedHistory: { time: string; duration: number; stack: string; resource: string }[] = [];
```

Comment:
From the "blocked-at" docs:

> The stack trace is pointing to a start of a function called asynchronously, so in most cases the first stack frame pointing to your code is where you need to start analyzing all synchronous operations to find the slow one.
> 
> In some cases your code is not directly called and tracking it down will still be difficult.

This concerns me a bit. I would find it very helpful if it had the sync call stacks, but just the async ones feels like it may mislead some users and still not have the important info. The other piece that concerns me is that by nature the warning message can't be sent until _after_ the sync code finishes, meaning if your code hangs indefinitely you won't see these logs.

Have you considered these drawbacks and do you think this is still worth it?

Just FYI @liliankasem is in the early stages of addressing this same situation with a "worker heartbeat" feature. It may or may not work in the case of Node.js, but it is another avenue to explore: https://dev.azure.com/msazure/One/_git/AAPT-Antares-Docs/pullrequest/6035812
(If you have any CRIs where a user had problems with blocking sync code please forward those along to Lilian)

Reply:
Yes, I considered these drawbacks. The actual call stacks were not very useful but it's better then nothing, also it includes "resource" of the blocking. Anyway I would not document the feature and only tell about it during CRIs investigations as the feature enabling can impact the perf. There is another lib which does not impact the perf but does not have the stack traces: https://www.npmjs.com/package/blocked. This feature can be constantly enabled. Do we think it will work better?

Regarding @liliankasem's doc we scheduled a meeting. actually we already have "WrkerStatusRequest" feature which